crypto/dilithium/p_dilithium3_asn1.c

```diff
@@ -11,138 +11,140 @@
 #include "../fipsmodule/evp/internal.h"
 #include "../internal.h"
 #include "sig_dilithium.h"
+#include "internal.h"
 
-
-static void dilithium3_free(EVP_PKEY *pkey) {
-  DILITHIUM3_KEY *key = pkey->pkey.ptr;
-  if (key == NULL) {
-    return;
-  }
-  OPENSSL_free(key->pub);
-  key->pub = NULL;
-  OPENSSL_free(key->priv);
-  key->priv = NULL;
-  OPENSSL_free(pkey->pkey.ptr);
-  pkey->pkey.ptr = NULL;
+static void pqdsa_free(EVP_PKEY *pkey) {
+  PQDSA_KEY_free(pkey->pkey.pqdsa_key);
+  pkey->pkey.pqdsa_key = NULL;
 }
 
-static int dilithium3_set_priv_raw(EVP_PKEY *pkey, const uint8_t *privkey,
+static int pqdsa_set_priv_raw(EVP_PKEY *pkey, const uint8_t *privkey,
         size_t privkey_len, const uint8_t *pubkey, size_t pubkey_len) {
-  if (privkey_len != DILITHIUM3_PRIVATE_KEY_BYTES) {
-    OPENSSL_PUT_ERROR(EVP, EVP_R_DECODE_ERROR);
+
+  PQDSA_KEY *key = OPENSSL_malloc(sizeof(PQDSA_KEY));
+  if (key == NULL) {
     return 0;
   }
 
-  // At time of writing, all |set_priv_raw| and |dilithium3_set_priv_raw|
+  // At time of writing, all |set_priv_raw| and |pqdsa_set_priv_raw|
   // invocations specify NULL public key. If that changes, we should modify
   // the conditional below to set the public key on |key|.
   if (pubkey != NULL) {
     OPENSSL_PUT_ERROR(EVP, EVP_R_OPERATION_NOT_SUPPORTED_FOR_THIS_KEYTYPE);
     return 0;
   }
 
-  DILITHIUM3_KEY *key = OPENSSL_malloc(sizeof(DILITHIUM3_KEY));
-  if (key == NULL) {
-      return 0;
-  }
-  key->priv = OPENSSL_malloc(DILITHIUM3_PRIVATE_KEY_BYTES);
-  if (key->priv == NULL) {
-    OPENSSL_free(key);
-    return 0;
-  }
-
-  key->pub = NULL;
-  OPENSSL_memcpy(key->priv, privkey, privkey_len);
-
-  dilithium3_free(pkey);
-  pkey->pkey.ptr = key;
+  // NOTE: No checks are done in this function, the caller has to ensure
+  //       that the pointers are valid and |in| has the correct size.
```

Comment:
what's `|in|`?

Reply:
Fixed comment in [c72e3e6](https://github.com/aws/aws-lc/pull/1963/commits/c72e3e6d667a865c8828034cf8c72cd81fa24bd1)